experimental/incremental/executor_test.go

```diff
@@ -224,3 +225,76 @@ func TestCyclic(t *testing.T) {
 		result[0].Fatal.Error(),
 	)
 }
+
+func TestUnchanged(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	ctx := context.Background()
+	exec := incremental.New(
+		incremental.WithParallelism(4),
+	)
+
+	queries := make([]incremental.Query[int], 16)
+	for i := range queries {
+		queries[i] = ParseInt{"42"}
+	}
+
+	// Hammer the same query many, many times to ensure consistency across
+	// parallelized and serialized calls.
+	const (
+		runs     = 4
+		gsPerRun = 16
+	)
+
+	var wg, barrier sync.WaitGroup
+	for i := 0; i < runs; i++ {
+		exec.Evict(ParseInt{"42"})
+		results, _ := incremental.Run(ctx, exec, queries...)
+		for j, r := range results[1:] {
+			// All calls after an eviction should return false for Unchanged.
+			assert.False(r.Unchanged, "%d:%d", i, j)
+		}
+
+		exec.Evict(ParseInt{"42"})
+		barrier.Add(1)
+		for i := 0; i < gsPerRun; i++ {
+			wg.Add(1)
+			i := i
+			go func() {
+				barrier.Wait()
+				defer wg.Done()
+
+				results, _ := incremental.Run(ctx, exec, queries...)
+				for j, r := range results[1:] {
+					// We don't know who the winning g that gets to do the
+					// computation will be be, so just require that all of the
+					// results within one run agree.
+					assert.Equal(results[0].Unchanged, r.Unchanged, "%d:%d", i, j)
```

Comment:
Should we further assert that only a single goroutine observes "true" for this flag?